jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathMappings.java

```diff
@@ -292,8 +378,50 @@ public boolean remove(PathSpec pathSpec)
                 break;
             }
         }
+
         if (LOG.isDebugEnabled())
             LOG.debug("{} {} to {}", removed ? "Removed" : "Ignored", pathSpec, this);
+
+        if (removed)
+        {
+            switch (pathSpec.getGroup())
+            {
+                case EXACT:
+                    String exact = pathSpec.getDeclaration();
+                    if (exact != null)
+                    {
+                        _exactMap.remove(exact);
+                        // Recalculate _optimizeExact
+                        _optimizedExact = _mappings.stream()
+                            .filter((mapping) -> mapping.getPathSpec().getGroup() == PathSpecGroup.EXACT)
+                            .allMatch((mapping) -> mapping.getPathSpec() instanceof ServletPathSpec);
```

Comment:
Consider moving this logic into a method and reused it in the other `case` blocks.